tests/test_ssh2.c

```diff
@@ -63,6 +92,24 @@ int main(int argc, char *argv[])
     }
 #endif
 
+#if defined(HAVE_PTHREAD_H)
+    pthread_t thread;
+
+    struct sigaction signal_handler_config = {
+        .sa_flags = SA_RESTART | SA_SIGINFO,
+        .sa_sigaction = signal_handler_function
+    };
+    sigemptyset(&signal_handler_config.sa_mask);
+
+    if(sigaction(SIGPROF, &signal_handler_config, NULL) != 0) {
```

Comment:
`sigaction` also non-portable.

Reply:
SIGPROF is the whole reason for this PR. It is used in Linux to gather process status. Failing to handle SIGPROF results in SSH connection failure rather than allowing the application using the library to be profiled by an external measurement tool.

It is OK to be non-portable so long as it's properly wrapped to prevent breaking on systems that don't have SIGPROF.

Because all systems (that I am aware of) with `SIGPROF` also have `sigaction`, I would also see no reason not to use `sigaction` HTH!

Reply:
We support more than just Linux. Also as I understand this is a causal example code committed to `tests`, but the essence of the feature is in `src`. CI runs fine now, but `pthread.h` detection is missing from autotools, and broken in CMake, so none of the example code is actually built.

Fixing autotools to have pthread detection might be one step forward, but it is complex. Even CMake is non-trivial if I look at how curl does these. We should also limit these detections to builds with tests enabled, otherwise it makes all builds slower. Only after implemeting these detection, could we see how the code compiles and runs in CI.